packages/react-components/react-tree/src/components/TreeProvider.tsx

```diff
@@ -0,0 +1,21 @@
+import * as React from 'react';
+import {
+  SubtreeContext,
+  SubtreeContextValue,
+  TreeContextValue,
+  TreeContext,
+  defaultRootTreeContextValue,
+} from '../contexts';
+
+export const TreeProvider = React.memo((props: React.ProviderProps<TreeContextValue | SubtreeContextValue>) => {
```

Comment:
`.memo()` there is useless as you have `children` and they will never have a stable reference